### PR TITLE
Release 15.20.3

### PR DESCRIPTION
**Pull request by augustas**
- _Fix_ [REL-783](https://oat-sa.atlassian.net/browse/REL-783)  Extract TAO cache setup

### CODE REVIEW

File: common/oatbox/cache/SetupFileCache.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    /**
+     * @throws Exception
+     */
+    public function createDirectory($cachePath): void
+    {
+        if (is_dir($cachePath) && is_writable($cachePath)) {
+            return;
+        }
+
+        if (!@mkdir($cachePath, 0700, true)) {
```
Comment (yaraslau-kavaliou) [inline common/oatbox/cache/SetupFileCache.php:41]: Do we really need to suppress errors using '@'?

Comment (augustas) [reply common/oatbox/cache/SetupFileCache.php:41]: Not essential, but makes the code shorter. Warnings themselves are not helpful in this case, an exception is thrown and logged if directory creation fails

Comment (yaraslau-kavaliou) [reply common/oatbox/cache/SetupFileCache.php:41]: I propose to do like this in order to not lose error message:
```
if (!@mkdir($cachePath, 0700, true)) {
    $error = error_get_last();
    throw new Exception(sprintf( 'Could not create application cache at "%s". Error message: %s', $cachePath, $error['message'] ?? 'unknown') );
}
```

Comment (augustas) [reply common/oatbox/cache/SetupFileCache.php:41]: I updated it, thanks for suggestion

**Review by yaraslau-kavaliou — APPROVED** [review]

**Pull request merged**